src/gui/org/deidentifier/arx/gui/view/impl/menu/DialogOrderSelection.java

```diff
@@ -303,19 +303,42 @@ protected void createButtonsForButtonBar(final Composite parent) {
             public void widgetSelected(final SelectionEvent e) {
                 String file = controller.actionShowOpenFileDialog(getShell(), "*.csv"); //$NON-NLS-1$
                 if (file != null){
-                    String[] array = loadArray(file, Charset.defaultCharset());
-                    if (array != null) {
-                        
-                        // Select string
-                        for (int i=0; i<combo.getItems().length; i++){
-                            if (combo.getItem(i).equals("String")) { //$NON-NLS-1$
-                                combo.select(i);
+                    ArrayList<String> fileData = loadFile(file, Charset.defaultCharset());
+                    long fileDataLines = fileData.size();
+
+                    if (fileDataLines != 0) {
+                        // remove values that are not present in the attribute's domain
+                        fileData.retainAll(Arrays.asList(elements));
+
+                        if (fileData.isEmpty() || fileData.size() != elements.length) {
+                            // import failed, file contains no or not all of the
+                            // attribute's domain
+                            controller.actionShowInfoDialog(getShell(),
+                                                               Resources.getMessage("DialogOrderSelection.16"),
+                                                               Resources.getMessage("DialogOrderSelection.17"));
+                        } else if (fileDataLines != fileData.size()) {
+                            // import failed, the file contains more values as
+                            // in the attribute's domain
+                            controller.actionShowInfoDialog(getShell(),
+                                                            Resources.getMessage("DialogOrderSelection.16"),
+                                                            Resources.getMessage("DialogOrderSelection.17"));
+                        } else {
+                            // Select string
+                            for (int i = 0; i < combo.getItems().length; i++) {
+                                if (combo.getItem(i).equals("String")) { //$NON-NLS-1$
+                                    combo.select(i);
+                                }
                             }
+
+                            // Set items
+                            elements = fileData.toArray(new String[fileData.size()]);
+                            list.setItems(elements);
                         }
-                        
-                        // Set items
-                        elements = array;
-                        list.setItems(array);
+                    } else {

```

Comment:
Does "file==null" really indicate an empty file? I think it indicates that the user has clicked cancel in the file dialog. What does the error message say? Is it actually needed? Wouldn't it be OK to silently exit the method, if no file was selected?